packages/zowe-explorer/src/extension.ts

```diff
@@ -265,28 +265,36 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
             )
         );
         context.subscriptions.push(
-            vscode.workspace.onDidSaveTextDocument(async (savedFile) => {
+            vscode.workspace.onWillSaveTextDocument(async (savedFile) => {
                 globals.LOG.debug(
                     localize(
                         "onDidSaveTextDocument1",
                         "File was saved -- determining whether the file is a USS file or Data set.\n Comparing (case insensitive) "
                     ) +
-                        savedFile.fileName +
+                        savedFile.document.fileName +
                         localize("onDidSaveTextDocument2", " against directory ") +
                         globals.DS_DIR +
                         localize("onDidSaveTextDocument3", "and") +
                         globals.USS_DIR
                 );
-                if (savedFile.fileName.toUpperCase().indexOf(globals.DS_DIR.toUpperCase()) >= 0) {
+
+                const autoSaveEnabled = vscode.workspace.getConfiguration("files").autoSave === "off";
+                if (!savedFile.document.isDirty && autoSaveEnabled) {
+                    globals.LOG.debug(
+                        localize("activate.didSaveText.file", "File ") +
+                            savedFile.document.fileName +
+                            localize("activate.didSaveText.notDirty", " is not a dirty file ")
```

Comment:
Would it be better if it exits the function if no saving needs to be done?

Reply:
Logging is a good idea, so I'm in favor of leaving it there. The code that follows this block handles saving different types of files, but if you don't want a save to happen anymore, should this block just exit? Exiting a function when no more work needs to happen is cleaner and results in less bugs.